packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,397 @@
+                    },
+                  })
+                ),
+                O.alt(() =>
+                  pipe(
+                    hoppRequestBody,
+                    (hoppRequestBody): OpenAPIV3.RequestBodyObject => ({
+                      content: {
+                        "text/plain": {
+                          schema: {
+                            type: "string",
+                            default: hoppRequestBody.body,
+                          },
+                        },
+                      },
+                    }),
+                    O.some
+                  )
+                )
+              )
+            ),
+            O.alt(() =>
+              pipe(
+                hoppRequestBody,
+                O.fromPredicate(isFormDataBody),
+                O.map(
+                  (body): OpenAPIV3.RequestBodyObject => ({
+                    content: {
+                      [contentType]: {
+                        schema: {
+                          properties: {
+                            ...generateFormDataEntries(body),
+                          },
+                        },
+                      },
+                    },
+                  })
+                )
+              )
+            ),
+            E.fromOption(() => "INVALID_BODY" as const)
+          )
+        )
+      )
+    )
+  )
+
+export const generateOpenApiAuth = (
+  hoppAuth: HoppRESTAuth
+): E.Either<"INVALID_AUTH", OpenAPIV3.SecurityRequirementObject> =>
+  pipe(
+    hoppAuth.authType,
+    O.fromPredicate((authType) => authType === "none"),
+    O.map(() => ({})),
+    O.alt(() =>
+      pipe(
+        hoppAuth.authType,
+        O.fromPredicate(
+          (authType): authType is "basic" | "api-key" | "bearer" =>
+            authType !== "none" && authType !== "oauth-2"
+        ),
+        O.map((authType) =>
+          pipe(
+            {
+              basic: "basicAuth",
+              "api-key": "ApiKeyAuth",
+              bearer: "bearerAuth",
+            },
+            (auths) => ({
+              [auths[authType]]: [],
+            })
+          )
+        )
+      )
+    ),
+    E.fromOption(() => "INVALID_AUTH" as const)
+  )
+
+const isValidMethod = (method: string): method is OpenAPIV3.HttpMethods =>
+  [
+    "get",
+    "post",
+    "patch",
+    "delete",
+    "options",
+    "head",
+    "patch",
+    "trace",
+  ].includes(method)
+
+const generateOpenApiDocument = (
+  paths: OpenAPIV3.PathsObject
+): OpenAPIV3.Document => ({
+  openapi: "3.0.0",
+  info: {
+    title: "Hoppscotch Openapi Export",
+    version: "1.0.0",
+  },
+  paths,
+  components: {
+    securitySchemes: {
+      basicAuth: {
+        type: "http",
+        scheme: "basic",
+      },
+      bearerAuth: {
+        type: "http",
+        scheme: "bearer",
+      },
+      ApiKeyAuth: {
+        type: "apiKey",
+        name: "api-key-header-name",
+        in: "header",
+      },
+    },
+  },
+})
+
+type PathGenerationErrors =
```

Comment:
Type names should be singular

```suggestion
type PathGenerationError =
```